nixos/modules/services/security/bitwarden_rs/default.nix

```diff
@@ -18,15 +18,33 @@ let
         else key + toUpper x) "" parts;
     in if builtins.match "[A-Z0-9_]+" name != null then name else partsToEnvVar parts;
 
-  configFile = pkgs.writeText "bitwarden_rs.env" (concatMapStrings (s: s + "\n") (
-    (concatLists (mapAttrsToList (name: value:
-      if value != null then [ "${nameToEnvVar name}=${if isBool value then boolToString value else toString value}" ] else []
-    ) cfg.config))));
+  # Due to the different naming schemes allowed for config keys,
+  # we can only check for values consistently after converting them to their corresponding environment variable name.
+  configEnv =
+    let
+      configEnv = listToAttrs (concatLists (mapAttrsToList (name: value:
+        if value != null then [ (nameValuePair (nameToEnvVar name) (if isBool value then boolToString value else toString value)) ] else []
+      ) cfg.config));
+    in { DATA_FOLDER = "/var/lib/bitwarden_rs"; } // optionalAttrs (!(configEnv ? WEB_VAULT_ENABLED) || configEnv.WEB_VAULT_ENABLED == "true") {
+      WEB_VAULT_FOLDER = "${pkgs.bitwarden_rs-vault}/share/bitwarden_rs/vault";
+    } // configEnv;
+
+  configFile = pkgs.writeText "bitwarden_rs.env" (concatStrings (mapAttrsToList (name: value: "${name}=${value}\n") configEnv));
+
+  bitwarden_rs = pkgs.bitwarden_rs.override { inherit (cfg) dbBackend; };
```

Comment:
Does this mean that somebody using bitwarden_rs with mysql support needs to recompile the package? I'd prefer to have e.g. `bitwarden_rs-mysql`, `bitwarden_rs-postgres` and `bitwarden_rs-sqlite` (which defaults to `bitwarden_rs-sqlite`) to ensure that everything's built by Hydra.

Reply:
Good suggestion! I have added those packages to `pkgs/top-level/all-packages.nix`. This should let Hydra prebuilt them AFAIK, since the line you commented on would produce the same derivation as those when supplied with one of the allowed backend names. I only changed your `postgres` to `postgresql` as that is how the feature flag for it is called in bitwarden_rs and how it is named in nixpkgs.